power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -622,21 +610,24 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
         initialize(regulator_order);
 
         if (auto result = iterate_with_fallback(state, regulator_order, method); strategy_ == OptimizerStrategy::any) {
-            return produce_output(state, std::move(result));
+            return produce_output(regulator_order, std::move(result));
         }
 
         // refine solution
         exploit_neighborhood(regulator_order);
-        return produce_output(state, iterate_with_fallback(state, regulator_order, method));
+        return produce_output(regulator_order, iterate_with_fallback(state, regulator_order, method));
     }
 
-    auto produce_output(State const& state, ResultType solver_output) const -> MathOutput<ResultType> {
+    auto produce_output(std::vector<std::vector<RegulatedTransformer>> const& regulator_order,
+                        ResultType solver_output) const -> MathOutput<ResultType> {
         TransformerTapPositionResult transformer_tap_positions;
 
-        // TODO(mgovers): only output the transformers that are regulated
-        (get_transformer_tap_positions<TransformerTypes, typename State::ComponentContainer>(state,
-                                                                                             transformer_tap_positions),
-         ...);
+        for (auto const& sub_order : regulator_order) {
+            for (auto const& regulator : sub_order) {
+                auto const& transformer = regulator.transformer;
+                transformer_tap_positions.push_back({transformer.id(), transformer.tap_pos()});
```

Comment:
```suggestion
                transformer_tap_positions.emplace_back({transformer.id(), transformer.tap_pos()});
```
on behalf of sonar cloud

Reply:
i now know why you initially used `push_back`. see https://developercommunity.visualstudio.com/t/stdvectoremplace-back-does-not-work-for-a-class-wi/1691516 and in particular this minimal example: https://godbolt.org/z/xovjh9ffe . 

looks like a compiler bug. we can change it in the future when it's resolved

Reply:
FYI: this was fixed in clang `16.0.0` https://godbolt.org/z/vYohvbv7q

Reply:
another reason to migrate to Ubuntu 24.04 runner images